src/metabolic_ninja/worker/designer.py

```diff
@@ -0,0 +1,285 @@
+# Copyright (c) 2018, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import cameo.core.target
+from cameo.strain_design import DifferentialFVA, OptGene
+from cameo.strain_design.heuristic.evolutionary.objective_functions import (
+    biomass_product_coupled_min_yield,
+    product_yield,
+)
+from cameo.strain_design.heuristic.evolutionary_based import CofactorSwapOptimization
+from cobra.exceptions import OptimizationError
+from numpy import isnan
+
+from .evaluate import evaluate_biomass_coupled_production, evaluate_production
+from .helpers import find_synthetic_reactions, manipulation_helper
+
+
+logger = logging.getLogger(__name__)
+
+
+def differential_fva_optimization(pathway, model):
+    """
+    Compare FVA results on the production plane with maximum growth.
+
+    Parameters
+    ----------
+    pathway :
+        A heterologous pathway identified by cameo.
+    model : cobra.Model
+        The model under investigation.
+
+    Returns
+    -------
+    cameo.design
+        A number of differential FVA designs corresponding to evenly spaced
+        points on the surface of the phenotypic phase plane.
+
+    """
+    with model:
+        pathway.apply(model)
+        predictor = DifferentialFVA(
+            design_space_model=model,
+            objective=pathway.product.id,
+            variables=[model.biomass],
+            normalize_ranges_by=model.biomass,
+            # Excluding the maxima, this corresponds to three evenly spaced
+            # designs.
+            points=5,
+        )
+        try:
+            designs = predictor.run(progress=False)
+        except ZeroDivisionError as error:
+            logger.warning(
```

Comment:
I wonder if this should be elevated to error level so it's visible in sentry and forces us to make a fix.

Reply:
Sounds reasonable to me. Can you create a PR or issue for that, please?

Reply:
https://github.com/DD-DeCaF/metabolic-ninja/pull/50